src/app/controllers/checkout/checkout.controller.ts

```diff
@@ -0,0 +1,210 @@
+      this.$http = $http;
+      this.$window = $window;
+      this.$state = $state;
+      this.$stateParams = $stateParams;
+
+      let userToken =  this.$window.localStorage.getItem('token');
+      if (!this.isLoggedIn()) {
+        this.$state.go("signup");
+      }
+
+      if(this.$state.$current.name = 'checkout') {
+        this.$state.go('checkout.stepOne');
+      }
+
+      this.getAddresses();
+      this.getProducts();
+      this.getCards();
+            
+    }
+
+    getProducts() : void {
+    
+      let userToken =  this.$window.localStorage.getItem('token');
+      if (this.isLoggedIn()) {
+        this.$http({
+            method  : 'GET',
+            url     :  this.rootUrl + '/cart',
+            headers :  {'x-authentication': userToken}
+          }).then((response: any) => {
+            this.productList = response.data.products;
+            this.itemsInCart = _.get(this.productList, 'length', 0);
+
+          }, (errorResponse: any) => {
+            alert('Erro: ' + _.get(errorResponse, 'data.message'));
+        });
+      }
+    }
+
+     getCards() : void {
+    
+      let userToken =  this.$window.localStorage.getItem('token');
+      if (this.isLoggedIn()) {
+        this.$http({
+            method  : 'GET',
+            url     :  this.rootUrl + '/card',
+            headers :  {'x-authentication': userToken}
+          }).then((response: any) => {
+            this.cardList = response.data.cards;
+
+          }, (errorResponse: any) => {
+            alert('Erro: ' + _.get(errorResponse, 'data.message'));            
+        });
+      }
+    }
+
+    getAddresses() : void {
+    
+      let userToken =  this.$window.localStorage.getItem('token');
+      if (this.isLoggedIn()) {
+        this.$http({
+            method  : 'GET',
+            url     :  this.rootUrl + '/address',
+            headers :  {'x-authentication': userToken}
+          }).then((response: any) => {
+            this.addressList = response.data.addresses;
+
+          }, (errorResponse: any) => {
+            alert('Erro: ' + _.get(errorResponse, 'data.message'));
+        });
+      }
+    }
+
+    chooseAddress(address: Address) : void {
+      this.chosenAddress = address;
+      this.$state.go("checkout.stepTwo");
+    }
+
+    chooseCard(card: Card) : void {
+      this.chosenCard = card;
+      this.getFreight();
+      this.$state.go("checkout.stepThree");
+    }
+
+    formatMoney(x:  number) : string {
+      return "R$" + Math.round(x * Math.pow(10,2) )/ Math.pow(10 ,2);
+    }
+
+    calculateTotal():  string {
+      let total:  number = 0;
+      _.forEach(this.productList, (product: any) => {
+        total += product.quantity * product.price;
+      });
+
+      return this.formatMoney(total);
+    }
+
+    calculateTotalWithFreight(): string {
+      let total:  number = 0;
+      _.forEach(this.productList, (product: any) => {
+        total += product.quantity * product.price;
+      });
```

Comment:
this methods are doing the same `calculateTotalWithFreight` `calculateTotal`, didn't you forget to add  the freight??